cli/commands/root.go

```
@@ -209,6 +229,94 @@ func wunderctlBinaryPath() string {
 	return path
 }
 
+func dataSourcesTelemetryMetrics(info *telemetry.MetricClientInfo) ([]*telemetry.Metric, error) {
+	wunderGraphDir, err := files.FindWunderGraphDir(_wunderGraphDirConfig)
+	if err != nil {
+		return nil, err
+	}
+	configFile := filepath.Join(wunderGraphDir, "generated", "wundergraph.config.json")
+	f, err := os.Open(configFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	dec := json.NewDecoder(f)
+	var wgConfig wgpb.WunderGraphConfiguration
+	if err := dec.Decode(&wgConfig); err != nil {
+		return nil, err
+	}
+	var metrics []*telemetry.Metric
+	if api := wgConfig.Api; api != nil {
+		if engineConfig := api.EngineConfiguration; engineConfig != nil {
+			for _, ds := range engineConfig.DatasourceConfigurations {
+				var dataSourceName string
+				var urlVariable *wgpb.ConfigurationVariable
+				switch ds.Kind {
+				case wgpb.DataSourceKind_STATIC:
+					// Not reported
+				case wgpb.DataSourceKind_REST:
+					dataSourceName = ds.Kind.String()
+					urlVariable = ds.GetCustomRest().GetFetch().GetUrl()
+				case wgpb.DataSourceKind_GRAPHQL:
+					dataSourceName = ds.Kind.String()
+					urlVariable = ds.GetCustomGraphql().GetFetch().GetUrl()
+				case wgpb.DataSourceKind_POSTGRESQL,
+					wgpb.DataSourceKind_MYSQL,
+					wgpb.DataSourceKind_SQLSERVER,
+					wgpb.DataSourceKind_MONGODB,
+					wgpb.DataSourceKind_SQLITE:
+
+					dataSourceName = ds.Kind.String()
+					urlVariable = ds.CustomDatabase.GetDatabaseURL()
+				default:
+					if err != nil {
```

Do you want to return an error when the datasource is not supported? I'm not sure if the err check makes sense in this case.

---

The idea was to make it evident for future developers, but an error might be too harsh. Would you prefer a warning message in the logs while handling the known data source types?